src/adapter/src/catalog/builtin.rs

```diff
@@ -2794,4 +2802,60 @@ mod tests {
 
         Ok(())
     }
+
+    // Make sure pg catalog views don't use unsigned ints.
+    #[tokio::test]
+    async fn test_pg_catalog_unsigned() -> Result<(), anyhow::Error> {
+        let catalog = Catalog::open_debug_sqlite(SYSTEM_TIME.clone()).await?;
+        let conn_catalog = catalog.for_system_session();
+
+        for view in BUILTINS::views()
+            .filter(|view| view.schema == PG_CATALOG_SCHEMA || view.schema == INFORMATION_SCHEMA)
+        {
+            let item = conn_catalog.resolve_item(&PartialObjectName {
+                database: None,
+                schema: Some(view.schema.to_string()),
+                item: view.name.to_string(),
+            })?;
+            let full_name = conn_catalog.resolve_full_name(item.name());
+            for col_type in item.desc(&full_name)?.iter_types() {
+                match &col_type.scalar_type {
+                    ScalarType::UInt16 | ScalarType::UInt32 | ScalarType::UInt64 => {
+                        return Err(anyhow!("Unsigned type found in {full_name}"))
+                    }
+                    ScalarType::Bool
+                    | ScalarType::Int16
+                    | ScalarType::Int32
+                    | ScalarType::Int64
+                    | ScalarType::Float32
+                    | ScalarType::Float64
+                    | ScalarType::Numeric { .. }
+                    | ScalarType::Date
+                    | ScalarType::Time
+                    | ScalarType::Timestamp
+                    | ScalarType::TimestampTz
+                    | ScalarType::Interval
+                    | ScalarType::PgLegacyChar
+                    | ScalarType::Bytes
+                    | ScalarType::String
+                    | ScalarType::Char { .. }
+                    | ScalarType::VarChar { .. }
+                    | ScalarType::Jsonb
+                    | ScalarType::Uuid
+                    | ScalarType::Array(_)
+                    | ScalarType::List { .. }
+                    | ScalarType::Record { .. }
+                    | ScalarType::Oid
+                    | ScalarType::Map { .. }
```

Comment:
Map types should probably be forbidden here as well, since they are one of our extensions.